src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ExtensionDataReader.cs

```diff
@@ -37,20 +37,22 @@ private enum ExtensionDataNodeType
         private int _attributeCount;
         private int _attributeIndex;
 
+        private static object prefixLock = new object();
+
 #pragma warning disable 0649
         private XmlNodeReader _xmlNodeReader;
 #pragma warning restore 0649
 
         private XmlObjectSerializerReadContext _context;
 
-        private static Dictionary<string, string> s_nsToPrefixTable;
+        private static Hashtable s_nsToPrefixTable;
```

Comment:
Why the non-generic variant?

Reply:
Thanks!

So recap:
`s_nsToPrefixTable` is safe in this way. The write to it is under a _lock_, the read is double-checked under the _lock_.
`s_prefixToNsTable` gets written to under a _lock_, while the read is _unlocked_ in `LookupNamespace`. So the "safeness" depends on when `LookupNamespace` gets called in relation to `GetPrefix`. To be on the safe side your cite and `Hashtable` seems to be correct.

So there should be a comment on why `Hashtable` is used instead of `Dictionary<string, string>`, but it's not so obvious.

BTW: the two statics could be `readonly`.

Reply:
While I understand the desire to make sure things are commented so the code is understood, I feel like understanding the behavior characteristics of the main framework collection types should be a given when working with corefx code. Certainly if someone would want to change this they should understand the differences in behavior of HashTable and whatever they are wanting to replace it with before making a change. Otherwise do we need to add a comment to usages of List<T> explaining it's being used because of constant time retrieval by index as opposed to LinkedList which is linear time but can have constant time removal (whereas List<T> is linear time with size with some usage patterns)?

Reply:
O(1) vs O(n) is obvious, no need for any comment. The relation to List and LinkedList is a bit wrong in this place. It's more generic collection type vs nongeneric type with same